pennylane_lightning/core/src/gates/Constant.hpp

```diff
@@ -145,6 +193,42 @@ namespace Pennylane::Gates::Constant {
         GeneratorOperation::DoubleExcitationPlus,
         "GeneratorDoubleExcitationPlus"},
 };
+
```

Comment:
Similarly here:
```suggestion
    typedef std::pair<ControlledGeneratorOperation, std::string_view> choose_a_name;
```

Reply:
Let's make use of `using` rather than `typedef`, since the former is fully template compatible, while the later is not.

Reply:
Yes, `clang-tidy` complained about that.

Reply:
Makes sense.